### ISSUE TITLE
Fix CI when PR is raised from fork

### ISSUE THREAD

**Issue by Derpius** [TAServers/sweps#11]
- The pull request CI currently doesn't trigger automatically for security reasons, as it would run in the context of the base branch
  - How do other repos deal with this?
- The pull request CI does not fail for linting/formatting issues present in the files from a fork
  - I'd imagine it's somehow running on our `master` branch, and not the downstream branch

In general the CI also just wants cleaning up to avoid the duplicate workflow runs on each PR that's opened.

We should also fix this upstream in the gmod template repository.

**Comment by Kilometres** [comment]
"Workflows on pull requests to public repositories from some outside contributors will not run automatically, and might need to be approved first. By default, all first-time contributors require approval to run workflows."[ - here](https://docs.github.com/en/actions/managing-workflow-runs/approving-workflow-runs-from-public-forks#about-workflow-runs-from-public-forks)

Looks like dave's fork just needs approval to run the CI.
![image](https://github.com/TAServers/sweps/assets/39900063/256a0789-bf94-4e01-87d1-c84d3ed08416)

**Comment by Derpius** [comment]
Oh i know about that, the issue is when it runs the pipeline is passing (and his code definitely shouldnt pass)

I think it’s checking out the branch he’s trying to merge into, and not his branch

**Comment by Kilometres** [comment]
Could it be the detached head causing this?
![image](https://github.com/TAServers/sweps/assets/39900063/0e541702-3aba-4afd-8e54-d38f2f4c7838)

(This will probably be my last comment on this as I know nothing about Github actions and don't want to bloat the issue)

**Comment by Derpius** [comment]
Looking at the CI for wiremod, which seems to work fine on pull requests, it also ends up in a detached head state
